.github/workflows/push.yml

```diff
@@ -0,0 +1,40 @@
+name: Check build
+
+on:
+  push:
+  workflow_dispatch:
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  builddebug:
+    name: Debug
+    uses: actions/checkout@v3
+    with:
+      jobname: Build
+      unittests: true
+      cmakeopts: >
+        -DCMAKE_BUILD_TYPE=Debug -DKPSR_WITH_DOXYGEN=true
+        -DKPSR_TEST_PERFORMANCE=true -DKPSR_WITH_SOCKET=true -DKPSR_WITH_CODE_METRICS=true
+
+  buildrelease:
+    needs: builddebug
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - name: Vanilla
+            zmq: "false"
+          - name: ZMQ
+            zmq: "true"
+    name: ${{ matrix.name }} Release
+    uses: actions/checkout@v3
+    with:
+      jobname: Build
+      unittests: true
+      cmakeopts: >
+        -DKPSR_WITH_DOXYGEN=false
+        -DKPSR_TEST_PERFORMANCE=false -DKPSR_WITH_CODE_METRICS=false
+        -DCMAKE_BUILD_TYPE=RelWithDebInfo
```

Comment:
Add this cmake option: -DKPSR_WITH_ZMQ=${{ matrix.zmq }}

Reply:
only in buildrelease?